test/functional/core/startup_spec.lua

```diff
@@ -460,6 +461,18 @@ describe('startup', function()
     ]])
   end)
 
+  it('-r works without --headless in PTY #23294', function()
+    skip(is_os('win'))
```

Comment:
Why doesn't this work on WIndows?

Reply:
It hangs even with `--headless`. Maybe a bug in PTY or libuv code.

Reply:
Oh, it seems to be `jobwait()` causing the problem here.

Reply:
Ah right, jobwait() has some quirks on Windows IIRC.